### PR TITLE
CASM-3594: Require linux_distribution and recipe_type for recipes

### PR DESCRIPTION
**Pull request by jack-stanek-hpe**
## Summary and Scope

This change makes the `linux_distribution` and `recipe_type` properties required for recipes in the IUF product manifest schema. These two properties are expected by IMS when creating recipes, and the existing `ims-load-artifacts` image requires these properties to be provided when creating an image.

## Issues and Related PRs

* Resolves [CASM-3594](https://jira-pro.its.hpecorp.net:8443/browse/CASM-3594)

### Tested on:

  * Local development environment

### Test description:

Run `cray-nls validate` against test manifests without either linux_distribution or recipe_type and ensure they do not validate

## Risks and Mitigations

Requires communication with COS team since they will need to update their IUF manifest

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by haasken-hpe** [comment]
FYI @jamesclough-hpe. Will this be a problem for COS to add these two properties to the IUF Product Manifest file?

**Review by haasken-hpe — APPROVED** [review]

File: .version [hunk 1]
```diff
@@ -1 +1 @@
-0.6.0
+0.7.0
```
Comment (ekamin-hpe) [inline .version:1]: Heads up as you are resolving the conflicts, my recent PR (#95) updated the version to 0.7.0, so you may want to go to 0.8.0 here. Similarly, I updated the helm chart to version 1.4.22, so probably update that to 1.4.23.

Comment (ekamin-hpe) [reply .version:1]: Oh, and my change updated the schema version to 0.4.0, so it would probably be wise to update the schema version to 0.5.0 as well.

**Comment by jamesclough-hpe** [comment]
> FYI @jamesclough-hpe. Will this be a problem for COS to add these two properties to the IUF Product Manifest file?

Shouldn't be, just need to know when to pull in the vendor lib so we can update the iuf manifest in the same PR. Just need to know what the values that should be paired with these new settings. We will create a JIRA and block it on the CASM-3594 ticket if Dean hasn't already.

**Comment by haasken-hpe** [comment]
> > FYI @jamesclough-hpe. Will this be a problem for COS to add these two properties to the IUF Product Manifest file?
> 
> Shouldn't be, just need to know when to pull in the vendor lib so we can update the iuf manifest in the same PR. Just need to know what the values that should be paired with these new settings. We will create a JIRA and block it on the CASM-3594 ticket if Dean hasn't already.

It probably makes sense to block on CASM-3668 as well. That's where I'll update the version of the cray-nls image used to validate your IUF Product Manifest file against the schema.

**Review by ekamin-hpe — APPROVED** [review]

File: .version (hunk 1 above)
Comment (ekamin-hpe) [reply .version]: Looks good to me.

**Pull request merged**